blockchain/blockchain.go

```diff
@@ -15,6 +15,11 @@ import (
 
 const lenOfByteSlice = 8
 
+var (
+	startingBlockNumber *uint64
```

Comment:
We can put all this in `Synchronizer`. I don't think we need `blockchain` to be aware of all this, in fact, I think it shouldn't be.

Reply:
Done